Add resource management profile spec

andrew-gardener opened the pull request:
Related to #335

ViktorHaag commented on profiles/resource_management/caliper-profile-resource-management-v1p1.html:
```diff
@@ -0,0 +1,457 @@
+      <dt>Supported actions</dt>
+      <dd><a href="#action-activated">Activated</a></dd>
+      <dd><a href="#action-added">Added</a></dd>
+      <dd><a href="#action-copied">Copied</a></dd>
```
Why do we need "Copied"? Wouldn't this just be a "Create" of a new entity with the same data as another?

andrew-gardener replied:
I guess that it would enrich the action with extra data? It isn't represented in the spec right now but maybe the usage of that action would be to have the object set to the original entity and a generated for the new entity.

ViktorHaag replied:
Got it.

ViktorHaag commented on profiles/resource_management/caliper-profile-resource-management-v1p1.html:
```diff
@@ -0,0 +1,457 @@
+      <dd><a href=
+        "https://www.imsglobal.org/sites/default/files/caliper/v1p1/caliper-spec-v1p1/caliper-spec-v1p1.html#person"
+      >Person</a></dd>
+      <dt>Supported actions</dt>
```
Might be useful to have:
* Purged
* Restored (or Undeleted)

Such that "Delete" means a potentially recoverable deletion of an entity; "Purged" is a definitely not recoverable deletion; and "Restored" or "Undeleted" means recovery/reinstatement of a once-deleted entity.

andrew-gardener replied:
👍 This makes a lot of sense and answers the first set of questions I had in https://github.com/IMSGlobal/caliper-central/issues/138#issuecomment-434867934

ViktorHaag commented:
> can a `develop-profile/ResourceManagement-cc138` branch be made so I can change the base to it instead of `develop`?

Done -- you can now down-merge from develop, and repoint your PR to `develop-profile/ResourceManagement-cc138` on this repo.

arwhyte commented:
Given our current definition of "Deleted" 

> Wipe out digitally. Inverse of Created.

The addition of "Purged" strikes me as redundant.

"Publish" needs to expressed in past tense: "Published"

Unsure of the value of "Restored."

ViktorHaag commented:
We definitely have a concept of something getting "soft deleted" (restorable) and "hard deleted" (not restorable) and only having "Deleted" makes navigating this challenging.

If we decide that "Deleted" means a hard-delete, do we have a verb for a soft-delete concept (and within that context, "Restored" then becomes more meaningful)?

"Deleted" and "Purged" are the words we typically use, but I'm cool with having other words for the concepts, if we can address them (I'm more concerned that we address them, than about the particular words used).

jonespm commented:
Comments from the call today

1. Remove Purged add Archived (As Deleted conveys the meaning of Purged and needed something for soft-delete)
2. Restored relates to Archived
3. Rename instances of Publish -> Published / Unpublish->Unpublished
4. Indicate Published/Unpublished are also related to view state changes. 

I think that was it.

ViktorHaag commented:
Thanks, @jonespm -- looks good.

I wonder if we also want to include a `Drafted` verb to go along with Publish/Unpublish, since the "Draft --> Publish --> [Unpublish]" workflow seems pretty widespread?

andrew-gardener commented:
- Replaced `Purged` with `Archived`
- Added inverse relation between `Archived` and `Restored` definitions
- Fixed typos (`Publish` to `Published`, `Unpublish` to `Unpublished`, etc)
- Changed definition of `Published`/`Unpublish` to one that better communicates that there is a view state change (`have (one's work) issued for publication` to `Prepare and issue for public distribution or sale`)

andrew-gardener commented:
> I wonder if we also want to include a Drafted verb to go along with Publish/Unpublish, since the "Draft --> Publish --> [Unpublish]" workflow seems pretty widespread?

My take is that an entity that has been created but not published would inherently be a draft. 

It might be easier to add a state parameter to `DigitalResource` to help track this since some systems automatically publish, automatically draft, or give the option to draft/publish a entity on creation (so you wouldn't be necessarily be required to send `Published`/`Drafted` events in addition to the `Created` event).

andrew-gardener commented:
Added user stories and updated formatting in a couple places

andrew-gardener commented:
Added user stories and updated formatting in a couple places
